glide-core/src/client/value_conversion.rs

```diff
@@ -287,6 +288,46 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // Used by ZRANDMEMBER when the WITHSCORES arg is passed.
+        // The server response can be a nil value, an empty array, a flat array of member-score pairs, or a two-dimensional array of member-score pairs.
+        // The server response scores can be strings or doubles. The conversions we do here are as follows:
+        //
+        // - if the server returned nil, return nil
+        // - if the server returned an empty array, return an empty array
+        // - otherwise, return a two-dimensional array of member-score pairs, where scores are of type double
+        ExpectedReturnType::ArrayOfMemberScorePairs => {
+            match value {
+                Value::Nil => Ok(value),
+                Value::Array(ref array)
+                    if array.is_empty() || matches!(array[0], Value::Array(_)) =>
+                {
+                    // The server response is an empty array or a RESP3 array of pairs, with scores of type double,
+                    // so it is already in the correct format.
+                    Ok(value)
+                }
+                Value::Array(mut array)
+                    if array.len() > 1
+                        && matches!(array[1], Value::BulkString(_) | Value::SimpleString(_)) =>
```

Comment:
should we also check that this array is of even size?

Reply:
Sure, I can add that